src/components/fargekategori/fargekategori-popover.tsx

```diff
@@ -38,29 +38,32 @@ export const FargekategoriPopover = ({
     const veilederIdent = useSelectGjeldendeVeileder();
     const delay = (ms: number) => new Promise(resolve => setTimeout(resolve, ms));
     const handleOppdaterFargekategori = async (fargekategori: FargekategoriModell) => {
-        const data: FargekategoriDataModell = {
-            fnr: fnrs,
-            fargekategoriVerdi: fargekategori
-        };
+        if (enhet != null) {
+            const data: FargekategoriDataModell = {
+                fnr: fnrs,
+                fargekategoriVerdi: fargekategori,
+                enhetId: enhet
+            };
 
-        const apiResponseAction = await oppdaterFargekategoriAction(data)(dispatch);
+            const apiResponseAction = await oppdaterFargekategoriAction(data)(dispatch);
 
-        if (apiResponseAction?.type === FARGEKATEGORI_OPPDATER_OK && !apiResponseAction.data.errors.length) {
-            if (gammelFargekategori) {
-                const gammelStatustallId = fargekategoriUnderfilterKonfigurasjoner.find(
-                    konfigurasjon => konfigurasjon.filterId === gammelFargekategori
-                )?.statustallId;
-                const nyStatustallId = fargekategoriUnderfilterKonfigurasjoner.find(
-                    konfigurasjon => konfigurasjon.filterId === fargekategori
-                )?.statustallId;
-                await dispatch(leggTilStatustall(gammelStatustallId, -1));
-                await dispatch(leggTilStatustall(nyStatustallId, 1));
-            } else {
-                //Venter fordi det returneres FARGEKATEGORI_OPPDATER_OK før statustall er oppdatert i Opensearch
-                await delay(500);
-                dispatch(hentStatustallForVeileder(enhet, veilederIdent));
+            if (apiResponseAction?.type === FARGEKATEGORI_OPPDATER_OK && !apiResponseAction.data.errors.length) {
+                if (gammelFargekategori) {
+                    const gammelStatustallId = fargekategoriUnderfilterKonfigurasjoner.find(
+                        konfigurasjon => konfigurasjon.filterId === gammelFargekategori
+                    )?.statustallId;
+                    const nyStatustallId = fargekategoriUnderfilterKonfigurasjoner.find(
+                        konfigurasjon => konfigurasjon.filterId === fargekategori
+                    )?.statustallId;
+                    await dispatch(leggTilStatustall(gammelStatustallId, -1));
+                    await dispatch(leggTilStatustall(nyStatustallId, 1));
+                } else {
+                    //Venter fordi det returneres FARGEKATEGORI_OPPDATER_OK før statustall er oppdatert i Opensearch
+                    await delay(500);
+                    dispatch(hentStatustallForVeileder(enhet, veilederIdent));
+                }
+                setOpenState(false);
             }
```

Comment:
What happens when enhet is `null` though? Should we care about that now or just think that it's ok since you probably won't see anything in the table when you don't have `enhet`?